File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0
    _min = 0
    
    for l in sizes:
        if l[0] < l[1]:
            temp = l[0]
            l[0] = l[1]
            l[1] = temp
            
    for l in sizes:
        if l[0] > _min:
            _min = l[0]
    
    for l in sizes:
        if l[1] > answer:
            answer = l[1]
            
    answer = answer * _min
            
    return answer

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int answer = 0;
        int wmax = 0;
        int hmax = 0;
        
        for(int i = 0; i < sizes.length; i++) {
            Arrays.sort(sizes[i]);
        }
        
        for(int i = 0; i < sizes.length; i++) {
            if(sizes[i][0] > wmax)
                wmax = sizes[i][0];
            
            if(sizes[i][1] > hmax)
                hmax = sizes[i][1];
        }
        answer = wmax * hmax;
        
        return answer;
    }
}